### PR TITLE
Fix gauss constr

### LINKED ISSUE #162: Bug with GaussianConstraint
I want to add 3 constraints to a likelihood, see dummy example:

```python

const1 = zfit.Parameter("const1",  1.0)
const2 = zfit.Parameter("const2",  2.0)
const3 = zfit.Parameter("const3",  3.0)

constraint = {"params": [const1, const2, const3],
                        "mu": [1.0, 2.0, 3.0],
                        "sigma": [0.05, 0.05, 0.05]}

nll.add_constraints(zfit.constraint.GaussianConstraint(**constraint))

```
So this works but when I minimise the likelihood the parameters are not even close the the central value of the parameter and even if I put limits to the parameters this doesn't change anything (or the parameters are a the limits after the minimisation).

On the other hand If I do 

```python

const1 = zfit.Parameter("const1",  1.0)
const2 = zfit.Parameter("const2",  2.0)
const3 = zfit.Parameter("const3",  3.0)

constraint = {"params": [const1, const2],
                        "mu": [1.0, 2.0],
                        "sigma": [0.05, 0.05]}

nll.add_constraints(zfit.constraint.GaussianConstraint(**constraint))
nll.add_constraints(zfit.constraint.GaussianConstraint(params=const3, mu=const3, sigma=0.05))
```
It works perfectly ... the parameters are very close the central values of the constraints.
I have not clue yet why this happens.

## Context (Environment)

* zfit version: 0.3.3
* Python version: 3.6.7
* Are you using conda, pipenv, etc? : Conda
* Operating System: MacOS
* Tensorflow version: 1.13.1

### PR DESCRIPTION
**Pull request by mayou36**
Fixes #162 

## Proposed Changes

  -

## Tests added

  - test for three parameters and Gaussian constraint
  
## Checklist

 - [x] change approved
 - [x] implementation finished
 - [x] correct namespace imported
 - [x] tests added
 - [x] CHANGELOG updated

### CODE REVIEW

File: tests/test_constraint.py [hunk 1]
```diff
@@ -58,6 +59,76 @@ def test_gaussian_constraint():
...
+
+    constraint = {"params": [luminosity, syst1, syst2, syst3, efficiency],
+                  "mu": [1500, 1.0, 1.0, 1.0, 0.5],
+                  "sigma": [0.05 * 1500, 0.001, 0.01, 0.1, 0.05 * 0.5]}
+
+    constr1 = zfit.constraint.GaussianConstraint(**constraint)
+    # param_vals = [1500, 1.0, 1.0, 1.0, 0.5]
+    constraint['params'] = zfit.run(constraint['params'])
+
+    true_val = true_gauss_constr_value(**constraint)
```
[the remark this answers is not in the document]
Comment (mayou36) [reply tests/test_constraint.py]: no, bug2 errors...

**Comment by mayou36** [comment]
> It also seems that the order of the params, mu and sigma list is not preserved as shown in the example I sent you. Maybe a test checking that this order is kept can be added.

This would explain a lot, but I do not see where it gets messed up or how...

File: zfit/core/constraint.py [hunk 2]
```diff
@@ -184,10 +183,11 @@ def __init__(self, params: ztyping.ParamTypeInput, mu: ztyping.NumericalScalarTy
 
         self._covariance = covariance
         self._mu = mu
+        self._params_array = params
 
         distribution = tfd.MultivariateNormalFullCovariance
         dist_params = dict(loc=mu, covariance_matrix=covariance)
-        dist_kwargs = dict(validate_args=True)
+        dist_kwargs = dict(validate_args=False)
```
Comment (marinang) [inline zfit/core/constraint.py]: Why remove this check except for possible improvement in runtime performance?

Comment (mayou36) [reply zfit/core/constraint.py]: Good catch, thx! Forgot to change that back (done now)

**Comment by mayou36** [comment]
So I found it, and added a fix. I am not fully happy, maybe we need to restructure the Constraints anyway (e.g. also name? Make sure it's clear that it's an NLL constraint? But other topic)

**Comment by mayou36** [comment]
Ok, fixed, @marinang can you approve if ok?

File: zfit/core/constraint.py [hunk 3]
```diff
@@ -124,7 +124,6 @@ def __init__(self, params: Dict[str, ZfitParameter], distribution: tfd.Distribut
         self._distribution = distribution
         self.dist_params = dist_params
         self.dist_kwargs = dist_kwargs if dist_kwargs is not None else {}
-        self._tparams = tf.stack(self.get_params())
```
Comment (marinang) [inline zfit/core/constraint.py]: So this was the source of the problem, so `get_params` doesn't conserve the order ?

Comment (mayou36) [reply zfit/core/constraint.py]: Yes, not insertion order at least, not guaranteed...

**Pull request merged**